jni_gen/lib/src/config/config_provider.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:yaml/yaml.dart';
+
+class ConfigError extends Error {
+  ConfigError(this.message);
+  final String message;
+  @override
+  String toString() => message;
+}
+
+class ConfigProvider {
```

Comment:
What are the differences between this approach and https://github.com/dart-lang/ffigen/blob/master/lib/src/config_provider/config.dart#L224 ?

It could make sense to use some of the abstractions from FFIgen.

P.S. It is fine to copy code from FFIgen (such as `Specification`), both packages have the same license.

I've filed an issue to make the package FFIgen config_provider more reusable.